### ISSUE TITLE
gb/cmd: TestTestExtraFlags fails intermittently 

### ISSUE THREAD

**Issue by davecheney** [constabulary/gb#128]
```
--- FAIL: TestTestExtraFlags (0.00s)
        testflag_test.go:44: TestExtraFlags([-q -debug package_name]): want ([-q package_name],[-debug]), got ([package_name -q],[-debug])
FAIL
FAIL    github.com/constabulary/gb/cmd  7.045s
```

Most likely due to map ordering

**Comment by davecheney** [comment]
@demizer please fix this before merging #124

**Comment by demizer** [comment]
Hi. I think this is fixed in #124. I believe this version of the code contained bad logic. I have ran all of the tests over 30 times using my testflag branch and haven't been able to reproduce.

**Comment by davecheney** [comment]
Cool. Thanks. I'll merge #124
